CefSharp.WinForms.Example/Handlers/LifeSpanHandler.cs

```diff
@@ -69,17 +69,30 @@ void ILifeSpanHandler.OnAfterCreated(IWebBrowser browserControl, IBrowser browse
         bool ILifeSpanHandler.DoClose(IWebBrowser browserControl, IBrowser browser)
         {
             //The default CEF behaviour (return false) will send a OS close notification (e.g. WM_CLOSE).
-            //See the doc for this method for full details.    
-            // Allow devtools to close
+            //See the doc for this method for full details.
+
+            var windowHandle = browser.GetHost().GetWindowHandle();
+            var parentHandle = Control.FromChildHandle(windowHandle);
+            var chromiumWebBrowser = (ChromiumWebBrowser)browserControl;
+
+            //Check devtools is in seperate window or docked in the same window
             if (browser.MainFrame.Url.Equals("chrome-devtools://devtools/devtools_app.html"))
             {
+                //If the IBrowserHost.CloseDevTools() is called whilst the brower is docked
+                //we need to handle it appropriately.
+                //The only way to release the handle is by disposing of its parent.
+                if (parentHandle != null)
+                {
+                    chromiumWebBrowser.Invoke(new Action(() =>
+                    {
+                        parentHandle.Dispose();
```

Comment:
Are you actually calling `Dispose` on the `Panel` a second time here?

Reply:
I added this as a precaution in the event that if the `IBrowserHost.CloseDevTools()` was called directly without the use of `CloseDevToolsDocked()`. 
if `CloseDevToolsDocked()` gets called accordingly the `parentHandle` should already be null and the `Dispose` shouldn't get called for a second time. 
However in this instance the `SplitContainer.Panel2` will remain visible but all children  should be disposed.

I'm happy to change this so that all the clean up is done via the `ILifeSpanHandler.DoClose`

Reply:
Maybe in hindsight I should of also checked if the parent is `IsDisposed` or `Disposing`

Reply:
Ah I remember now, when calling to `CloseDevToolsDocked()` which will `Dispose` the `Panel`.
The `ILifeSpanHandler.DoClose()` does not get called at all, it goes straight to `ILifeSpanHandler.OnBeforeClose()`